app/javascript/components/Game.jsx

```diff
@@ -9,27 +9,25 @@ import { Switcher } from "./Switcher";
 import { Category } from "./Category";
 import { useError } from "./helpers/useError";
 import { Error } from "./Error";
+import { useSetUser } from "./helpers/useSetUser";
 
 export const Game = () => {
 
-    const id = useParams().id
+    const game_id = useParams()?.game_id
+    const { user, setUser, loading, setLoading, error, setError } = useSetUser()
```

Comment:
yeah, I may need to read more about the purpose of custom hooks in a future PR. The setters are mostly retrieved from the hook so that child components can use them to set the state of that hook (you're right--all those attributes are being passed so that any child component rendered as an `Outlet` can use them).

I think I was using this hook to also have a common shared loading/error state that I could just share among the components, but maybe that's the wrong way to think about hooks, and I should just hold that `error `and `loading` state in a parent component instead